dockers/docker-database/base_image_files/sonic-db-cli

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```

Comment:
>bash [](start = 7, length = 4)

I guess writing whole script in python is much easier.

Reply:
rewrite with python

Reply:
Still suggest package it into swsssdk. So that we don't need to change buildimage.

---
In reply to: [333795539](https://github.com/Azure/sonic-buildimage/pull/3590#discussion_r333795539) [](ancestors = 333795539)

Reply:
I don't think it should be part of swsssdk pkg. 

 1. sonic-db-cli has the same role as a command like "cp, ls, redis-cli", and should be executable. It is not a library function/class, it should considered as a application which is using swsssdk lib.

2. As we discussed earlier when modifying swsssdk API, python wheel distribute cannot copy file to the  location out of the /usr/local/lib/python2.7/dist-package directory. So even we make it part of swsssdk pkg, we cannot distribute it to /usr/bin/** under today's code base.

So I think current location is OK for this sonic-db-cli.

Reply:
I agree sonic-db-cli is an executable.
If you use [https://docs.python.org/3/distutils/setupscript.html#installing-scripts](https://docs.python.org/3/distutils/setupscript.html#installing-scripts), the script will be copied to `/usr/local/bin/`, which fit your target.

---
In reply to: [334212868](https://github.com/Azure/sonic-buildimage/pull/3590#discussion_r334212868) [](ancestors = 334212868)

Reply:
> I agree sonic-db-cli is an executable.
> If you use https://docs.python.org/3/distutils/setupscript.html#installing-scripts, the script will be copied to `/usr/local/bin/`, which fit your target.
> 
> In reply to: [334212868](https://github.com/Azure/sonic-buildimage/pull/3590#discussion_r334212868) [](ancestors = 334212868)

Now assuming sonic-db-cli is able to copied to /usr/bin/, this raise another question.
 If sonic-db-cli  belongs to swsssdk package, it make no sense for the other dockers to install it. Sonic-db-cli is targeted for host (using `docker exec -it database XXXX`), not for other dockers. When installing swsssdk, other dockers will have sonic-db-cli, which is there but not usable.

Reply:
Thanks for the great insight. It is a true requirement for accessing the same functionality inside docker container.
Example: dockers/docker-dhcp-relay/wait_for_intf.sh.j2 need run `redis-cli` inside dhcp-relay container.

So we could make this script fit both environment. If you use python `redis` library instead of running 'redis-cli` command, it should be easy.

---
In reply to: [334652534](https://github.com/Azure/sonic-buildimage/pull/3590#discussion_r334652534) [](ancestors = 334652534)

Reply:
@qiluo-msft  move this PR to swsssdk pkg, https://github.com/Azure/sonic-py-swsssdk/pull/54